src/app/projects/project-list/project-list.component.ts

```diff
@@ -238,6 +241,10 @@ export class ProjectListComponent implements OnInit, OnDestroy {
             this.storageService.state.projList.tableParams = this.tableParams;
           }
 
+          // We need to clone filters, not reference
+          this.previousFilters = JSON.parse(JSON.stringify(this.filterForAPI));
```

Comment:
What is happening here? Why stringily and then parse it again?

Reply:
Basically, it's a lazy clone. Setting previous filters will maintain a reference if it's not cloned, which means previous will always equal current. This gets around that issue by reverting the object to a string, and back again.

Reply:
Ahh, okay the spread operator crates a copy of an object or array. `this.previousFilters - { ...this.filterForApi }`